packages/router/build.mts

```diff
@@ -0,0 +1,31 @@
+import { build, defaultBuildOptions } from '@redwoodjs/framework-tools'
+import { writeFileSync } from 'node:fs'
+
+// ESM build
+await build({
+  buildOptions: {
+    ...defaultBuildOptions,
+    tsconfig: 'tsconfig.build.json',
+    format: 'esm',
+    packages: 'external',
+  },
+})
+
+// CJS build
+await build({
+  buildOptions: {
+    ...defaultBuildOptions,
+    tsconfig: 'tsconfig.build-cjs.json',
+    outdir: 'dist/cjs',
+    packages: 'external',
+  },
+})
+
+// Place a package.json file with `type: commonjs` in the dist folder so that
+// all .js files are treated as CommonJS files.
+writeFileSync('dist/cjs/package.json', JSON.stringify({ type: 'commonjs' }))
```

Comment:
Question here josh for my understanding. If you do this won't it mean dependencies won't be installed since no dependencies are listed here?

Reply:
@dac09 the published package tarball will have three `package.json` files. One in the root, one at `./dist/package.json` and one at `./dist/cjs/package.json`. The root package.json still has all the dependencies and is the one that tells npm/yarn what other packages to install.

We use this same pattern for `@redwoodjs/auth`. You can download that tarball and take a look if you're interested :) https://registry.npmjs.org/@redwoodjs/auth/-/auth-8.0.0-canary.710.tgz